spec.html

```diff
@@ -9232,7 +9232,7 @@ <h1>Static Semantics: IsIdentifierRef ( ): a Boolean</h1>
     <emu-clause id="sec-runtime-semantics-namedevaluation" oldids="sec-grouping-operator-runtime-semantics-namedevaluation,sec-function-definitions-runtime-semantics-namedevaluation,sec-arrow-function-definitions-runtime-semantics-namedevaluation,sec-generator-function-definitions-runtime-semantics-namedevaluation,sec-asyncgenerator-definitions-namedevaluation,sec-class-definitions-runtime-semantics-namedevaluation,sec-async-function-definitions-runtime-semantics-namedevaluation,sec-async-arrow-function-definitions-runtime-semantics-namedevaluation" type="sdo">
       <h1>
         Runtime Semantics: NamedEvaluation (
-          _name_: unknown,
+          _name_: a String,
```

Comment:
It looks like this can be Private Name as well, since it is passed `functionObject.[[ClassFieldInitializerName]]` in [EvaluateBody](https://tc39.es/ecma262/#sec-runtime-semantics-evaluatebody). Possibly all of a String, a Symbol, a Private Name, or `~empty~`?

Reply:
or `a property key, a Private Name, or ~empty~`?

Reply:
I'm pretty sure that `_name_` can't be `~empty~`. (And if it can, that's probably an editorial error, since that would entail passing `~empty~` to `SetFunctionName`, which would not end well.)

(When `EvaluateBody` invokes `NamedEvaluation` with `functionObject.[[ClassFieldInitializerName]]`, that's preceded by an assertion that `_functionObject_.[[ClassFieldInitializerName]] is not ~empty~`.)

But yes re Symbol and Private Name.